src/System.Text.RegularExpressions/tests/PrecompiledRegexScenarioTest.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using RegexTestNamespace;
+using Xunit;
+
+namespace System.Text.RegularExpressionsTests
+{
+    public class PrecompiledRegexScenarioTest
+    {
+        [Fact]
+        public void TestPrecompiledRegex()
+        {
+            string text = "asdf134success1245something";
+            RegexTestClass testClass = new RegexTestClass();
+
+            Assert.Equal(1, testClass.Matches(text).Count);
+            Assert.Equal(1, testClass.Match(text).Groups[0].Captures.Count);
+            Assert.Equal(text, testClass.Match(text).Groups[0].Value);
+        }
+    }
+}
+
+namespace RegexTestNamespace
```

Comment:
How was this generated? Will some kind of tool be released that is able to produce these precompiled regexes that set the `Caps` and `CapNames` properties for .NET Core instead of the `caps` and `capnames` fields in the full framework?